api/controller.py

```diff
@@ -625,9 +626,20 @@ def load_lane(self, lane_identifier):
             elif isinstance(lane, WorkList):
                 lane.children = [self._db.merge(child) for child in lane.children]
         else:
-            lane = get_one(
-                self._db, Lane, id=lane_identifier, library_id=library_id
-            )
+            try:
+                lane_identifier = int(lane_identifier)
+            except ValueError, e:
+                pass
+
+            if isinstance(lane_identifier, int):
+                lane = get_one(
+                    self._db, Lane, id=lane_identifier, library_id=library_id
+                )
+
+        if lane and not lane.visible_to(self.request_patron):
```

Comment:
This is why I had to add `request_patron`. Authentication is not required for any controller that uses `load_lane`, so `authenticated_patron_for_request` probably hasn't been called yet. I need a yes-or-no answer as to whether there is an authenticated patron or not, and `authenticated_patron_for_request` can return all sorts of things.